lib/graphql/schema/default_type_error.rb

```diff
@@ -0,0 +1,19 @@
+module GraphQL
+  class Schema
+    module DefaultTypeError
+      def self.call(value, field, parent_type, query_ctx)
+        if field.type.kind.non_null?
```

Comment:
That's a great idea! I was trying to think about what to inject, and it didn't strike me to use an error object. That will make it much easier for people to tell "what went wrong" AND it will make it easier to send new info into the function in a non-breaking way in the future!